GTG/backends/bugzilla/bugzilla.py

```diff
@@ -0,0 +1,201 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2009 - Guillaume Desmottes <gdesmott@gnome.org>
+#
+# This program is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import re
+import threading
+import socket
+
+from xmlrpc.client import Fault as XmlrpcFault
+
+from collections import namedtuple
+from urllib.parse import urlparse
+
+from gi.repository import GObject
+
+from GTG.core.translations import _
+from GTG.tools.logger import Log
+
+from GTG.backends.bugzilla import exceptions
+from GTG.backends.bugzilla.notification import send_notification
+from GTG.backends.bugzilla.services import create_bugzilla_service
+
+__all__ = ('BugInformationSyncTask',)
+
+BugSyncTaskInfo = namedtuple('BugSyncTaskInfo',
+                             ['scheme', 'net_location', 'bug_id'])
+BUG_URL_PATTERN = re.compile(r'^(https?)://(.+)/show_bug\.cgi\?id=(\d+)$')
+
+BUGZILLA_NO_PERM_MESSAGE = _("You have no permission to read bug <b>%s</b>.")
+BUGZILLA_INVALID_MESSAGE = _("Bug ID <b>%s</b> is invalid")
+BUGZILLA_NOT_EXIST_MESSAGE = _("Bug ID <b>%s</b> does not exist.")
+BUGZILLA_BUG_SYNC_FAIL_MESSAGE = _("Failed to synchronize information for "
+                                   "bug <b>%s</b> due to some unknown reason.")
+
+
+def parse_bug_url(url):
+    '''
+    Extract URL data
+
+    @param url: a string representing a URL
+    @return: a tuple containing scheme, hostname, and a list of key-value
+             queries
+    '''
+    r = urlparse(url)
+    queries = dict([item.split('=') for item in r.query.split('&')])
```

Comment:
Do you mind to use https://docs.python.org/3.0/library/urllib.parse.html#urllib.parse.parse_qs instead?